Detectors/EMCAL/simulation/src/RawWriter.cxx

```diff
@@ -300,32 +301,21 @@ int RawWriter::carryOverMethod(const header::RDHAny* rdh, const gsl::span<char>
                                const char* ptr, int maxSize, int splitID,
                                std::vector<char>& trailer, std::vector<char>& header) const
 {
-  int offs = ptr - &data[0]; // offset wrt the head of the payload
-  // make sure ptr and end of the suggested block are within the payload
-  assert(offs >= 0 && size_t(offs + maxSize) <= data.size());
 
-  // Read trailer template from the end of payload
+  int bytesLeft = data.data() + data.size() - ptr;
+  int leftAfterSplit = bytesLeft - maxSize;
+
   gsl::span<const uint32_t> payloadwords(reinterpret_cast<const uint32_t*>(data.data()), data.size() / sizeof(uint32_t));
   auto rcutrailer = RCUTrailer::constructFromPayloadWords(payloadwords);
+  int trailerSize = rcutrailer.getTrailerSize() * sizeof(uint32_t);
+
+  int Margin = 0;
 
-  int sizeNoTrailer = maxSize - rcutrailer.getTrailerSize() * sizeof(uint32_t);
-  // calculate payload size for RCU trailer:
-  // assume actualsize is in byte
-  // Payload size is defined as the number of 32-bit payload words
-  // -> actualSize to be converted to size of 32 bit words
-  auto payloadsize = sizeNoTrailer / sizeof(uint32_t);
-  rcutrailer.setPayloadSize(payloadsize);
-  auto trailerwords = rcutrailer.encode();
-  trailer.resize(trailerwords.size() * sizeof(uint32_t));
-  memcpy(trailer.data(), trailerwords.data(), trailer.size());
-  // Size to return differs between intermediate pages and last page
-  // - intermediate page: Size of the trailer needs to be removed as the trailer gets appended
-  // - last page: Size of the trailer needs to be included as the trailer gets replaced
-  int bytesLeft = data.size() - (ptr - &data[0]);
-  bool lastPage = bytesLeft <= maxSize;
-  int actualSize = maxSize;
-  if (!lastPage) {
-    actualSize = sizeNoTrailer;
+  if (leftAfterSplit < trailerSize + Margin) {
```

Comment:
using Margin=0 means that you only the trailer carried to new page, is this what you want?

Reply:
We need to check with how it is implemented in the firmware what the margin is. For the moment I would assume only the trailer is moved to the page.

Reply:
@shahor02 As we learned from Martin now the firmware will only put the RCUtrailer as a whole on the a new page in case it would have to be split due to insufficient space on the current page, but no additional payload. So margin is indeed 0.

Reply:
OK, will merge it once full CI passed (unless you want to change trailer size calculation to constant).